src/device/specification.go

```diff
@@ -0,0 +1,26 @@
+package device
+
+import "fmt"
+
+func (d Device) RequestSpecification(client Client) (Device, error) {
+	resp, err := client.Specification(d)
+	if err != nil {
+		d.logger.Warn(fmt.Sprintf("Error loading specification for [%s]", d.ID.String()))
+		return d, err
+	}
+
+	spec, err := client.EvaluateSpecificationResponse(resp)
+	if err != nil {
+		d.logger.Warn(fmt.Sprintf("Error evaluating specification for [%s]", d.ID.String()))
+		return d, err
+	}
+
+	return spec, nil
+}
+
+func (d Device) LoadFromSpecification(spec Device) Device {
+	d.Description = spec.Description
+	d.Endpoints = spec.Endpoints
+	d.Name = spec.Name
+	return d
```

Comment:
i think we are missing some assignments here. most importantly `Host` and `Port`. Setting these will enable us to track the same device as it changes IPs and maintains the same MAC address.